share/wake/lib/system/http.wake

```diff
@@ -0,0 +1,199 @@
+package http
+
+from http export type HttpRequest
+
+# The http methods that may be called against a web server.
+export data HttpMethod =
+    HttpMethodGet
+    HttpMethodHead
+    HttpMethodPost
+    HttpMethodPut
+    HttpMethodDelete
+    HttpMethodConnect
+    HttpMethodOptions
+    HttpMethodTrace
+    HttpMethodPatch
+
+# The headers that may be assocated with an http request.
+#
+# These are often used to specify how to interpret the body or to provide authorization.
+tuple HttpHeader =
+    Name: String
+    Value: String
+
+# The description of a http request to send to a web server.
+tuple HttpRequest =
+    # The url of the function to be called
+    Url: String
+    # The method to be called on the destination
+    Method: HttpMethod
+    # The headers to be set on the request
+    Headers: List HttpHeader
+    # The body of the request
+    Body: String
```

Comment:
...on that note, why are these private to begin with?  There's not any validation nor transformation being done in the helper functions, and together they expose the entire surface area of the type, so even if we don't recommend people mess with this directly, neither do we gain anything from keeping it hidden.

Reply:
Latest commit makes things more wonky in the type now, worth taking a look and seeing if you still agree.

Mainly my thought is just that the API has a very specific public surface and I don't want people using anything else because I want to be able to manipulate the internals without a major refactor for folks who reached in to the data type

Something something "all observable effects of your api will eventually be depended on"

Reply:
Yeah, that's probably getting complex enough to warrant the indirection.  And if we don't see the structure which will eventually be merged as the final representation, then I do definitely agree that it's right to hide it even if it stayed simple for the MVP.